### DISCUSSION TITLE
How to validate HTML Input's native "step" attribute

### DISCUSSION THREAD

**Discussion by brmi** [react-hook-form/react-hook-form#2715] [Q&A]
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefstep

`<input step="0.01" type="number" placeholder="Enter amount" />`

Is there a way I can validate if the user has typed in a number that has the correct step?

For example, the user should not be able to submit 20.0001, as the step is 0.01. 

```
<input 
                  name="amount" 
                  type="number" 
                  placeholder="Enter amount"
                  ref={register({
                    required: "Please provide an amount to redeem",
                    step: 0.01,
                    validate: {
                      max: (value) => value && value <= 20 || "Amount is too high",
                      min: (value) => value && value >=1 || "Amount is too low"
                    }
                    
                })}
```

**Comment by bluebill1049** [comment]
can you share a codesandbox?

**Reply by brmi** [reply]
Never used it before - can you see this? https://mcld5.csb.app/

**Reply by bluebill1049** [reply]
perfect, the `step` is missing. I think you can still validate the input anyhow, and your validation looks valid too.

**Reply by brmi** [reply]
Okie - thanks for the quick response. I was just wondering if it was "built in".

**Reply by bluebill1049** [reply]
but you can use `min` and `max` tho.
